### PR TITLE
Add study's name and path in page's title

### PR DESCRIPTION
**Pull request by archendel**

### CODE REVIEW

File: src/components/study-container.js [hunk 1]
```diff
@@ -304,6 +305,92 @@ export function StudyContainer({ view, onChangeTab }) {
         }
     }, [studyUuid, loadTree]);
 
+    useEffect(() => {
+        const appName = 'GridStudy';
+        const MAX_TITLE_LENGTH = 106;
+
+        if (studyUuid) {
+            fetchElementAndParentsInfo(studyUuid)
+                .then((response) => {
```
Comment (laurgarn) [inline src/components/study-container.js:314]: May be separate gathering of the response from building of the title from the response, in its own function.

File: src/components/study-container.js [hunk 2]
```diff
@@ -304,6 +305,107 @@ export function StudyContainer({ view, onChangeTab }) {
...
+    const computePageTitleWithFirstDirectory = (pageTitle, parents) => {
+        return pageTitle + (parents.length > 1 ? '...' : '') + '/' + parents[0];
+    };
+
+    const computePageTitleWithFullPath = (
+        pageTitle,
+        parents,
+        maxTitleLength
+    ) => {
+        let maxAllowedPathSize =
```
Comment (laurgarn) [inline src/components/study-container.js:317]: `const` is nicer (except in typing length :-/) than `let`

**Review by laurgarn — CHANGES_REQUESTED** [review]

File: src/components/study-container.js [hunk 3]
```diff
@@ -304,6 +306,35 @@ export function StudyContainer({ view, onChangeTab }) {
         }
     }, [studyUuid, loadTree]);
 
+    useEffect(() => {
+        const appName = 'GridStudy';
```
Comment (jonenst) [inline src/components/study-container.js:310]: get the original title from the html <title>

**Comment by JacquesBorsenberger** [comment]
can it be done with OverflowableText and   'direction: rtl;  text-align: left;'  in the css ? (should test on firefox&edge)

File: src/translations/en.json [hunk 4]
```diff
@@ -362,5 +362,7 @@
     "insertNetworkModificationNodeBefore": "Insert a network modification node before",
     "insertModelNodeBefore": "Insert a model node before",
     "insertNetworkModificationNodeAfter": "Insert a network modification node after",
-    "insertModelNodeAfter": "Insert a model node after"
+    "insertModelNodeAfter": "Insert a model node after",
+
+    "LoadStudyAndParentsInfoError": "An error occured while loading current study's metadata"
```
Comment (sBouzols) [inline src/translations/en.json:367]: metadata is something else, avoid confusion I think.

File: src/utils/compute-title.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+const MAX_TITLE_LENGTH = 106;
```
Comment (sBouzols) [inline src/utils/compute-title.js:1]: You need a license header in a new file

File: src/utils/compute-title.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        MAX_TITLE_LENGTH - pageTitle.length - '...'.length;
+
+    pageTitle = pageTitle + computePath(parents, maxAllowedPathSize);
+
+    return pageTitle;
+};
+
+export const computePageTitle = (appName, study, parents) => {
+    if (!study) return appName;
+    let pageTitle = appName + SEPARATOR + study.elementName;
```
Comment (sBouzols) [inline src/utils/compute-title.js:35]: You are dependent of the study Object format here. Seems a bad idea. Maybe this function should accept the studyName directly as string.

Comment (sBouzols) [reply src/utils/compute-title.js:35]: like parents which is an array of parents names

File: src/components/study-container.js [hunk 7]
```diff
@@ -304,6 +306,35 @@ export function StudyContainer({ view, onChangeTab }) {
...
+                        errorMessage: errorMessage,
+                        enqueueSnackbar: enqueueSnackbar,
+                        headerMessage: {
+                            headerMessageId: 'LoadStudyAndParentsInfoError',
+                            intlRef: intlRef,
+                        },
+                    });
+                });
+        } else {
+            document.title = appName;
```
Comment (sBouzols) [inline src/components/study-container.js:334]: maybe it's useless ? title is already set to something like 'GridStudy' elsewhere

File: src/components/study-container.js [hunk 8]
```diff
@@ -304,6 +306,35 @@ export function StudyContainer({ view, onChangeTab }) {
...
+                .then((response) => {
+                    const study = response[0];
+                    const parents = response
+                        .slice(1)
+                        .map((parent) => parent.elementName);
+
+                    document.title = computePageTitle(appName, study, parents);
+                })
+                .catch((errorMessage) => {
+                    document.title = appName;
```
Comment (sBouzols) [inline src/components/study-container.js:323]: remove

File: src/components/study-container.js [hunk 9]
```diff
@@ -304,6 +306,35 @@ export function StudyContainer({ view, onChangeTab }) {
...
+
+        if (studyUuid) {
+            fetchElementAndParentsInfo(studyUuid)
+                .then((response) => {
+                    const study = response[0];
+                    const parents = response
+                        .slice(1)
+                        .map((parent) => parent.elementName);
+
+                    document.title = computePageTitle(appName, study, parents);
```
Comment (sBouzols) [inline src/components/study-container.js:320]: `document.title = computePageTitle(document.title, studyName, parentsNames);` ?

**Review by sBouzols — CHANGES_REQUESTED** [review]
some comments for now, no tests

File: src/components/study-container.js (hunk 3 above)
Comment (archendel) [reply src/components/study-container.js]: fixed by setting the initial title in the component state

File: src/utils/compute-title.js (hunk 5 above)
Comment (archendel) [reply src/utils/compute-title.js:8]: fixed

File: src/components/study-container.js (hunk 8 above)
Comment (archendel) [reply src/components/study-container.js]: not removed to be able to come back to the default title if the study's uuid is changed and something wrong happens when loading it's name/path

File: src/translations/en.json (hunk 4 above)
Comment (archendel) [reply src/translations/en.json]: fixed

**Review by laurgarn — APPROVED** [review]

File: src/components/study-container.js (hunk 7 above)
Comment (sBouzols) [reply src/components/study-container.js]: it's useful when studyUuid change then ok

File: src/components/study-container.js (hunk 8 above)
Comment (sBouzols) [reply src/components/study-container.js]: ok

**Review by jonenst — APPROVED** [review]

**Review by sBouzols — APPROVED** [review]
Code review OK
Tests OK, Console checked no warnigns

**Pull request merged**